browser/playlist/playlist_service_factory.cc

```diff
@@ -53,26 +64,84 @@ class PlaylistServiceDelegateImpl : public PlaylistService::Delegate {
     auto tab_models = TabModelList::models();
     auto iter = base::ranges::find_if(
         tab_models, [](const auto& model) { return model->IsActiveModel(); });
-    if (iter == tab_models.end())
+    if (iter == tab_models.end()) {
       return nullptr;
+    }
 
     auto* active_contents = (*iter)->GetActiveWebContents();
     DCHECK(active_contents);
     DCHECK_EQ(active_contents->GetBrowserContext(), profile_.get());
     return active_contents;
 #else
     auto* browser = chrome::FindLastActiveWithProfile(profile_);
-    if (!browser)
+    if (!browser) {
       return nullptr;
+    }
 
     auto* tab_model = browser->tab_strip_model();
     DCHECK(tab_model);
     return tab_model->GetActiveWebContents();
 #endif  // BUILDFLAG(IS_ANDROID)
   }
 
+  void SanitizeImage(
+      std::unique_ptr<std::string> image,
+      base::OnceCallback<void(scoped_refptr<base::RefCountedBytes>)> callback)
+      override {
+    DecodeImageInIsolatedProcess(
+        std::move(image),
+        base::BindOnce(&PlaylistServiceDelegateImpl::EncodeAsPNG,
+                       weak_ptr_factory_.GetWeakPtr(), std::move(callback)));
+  }
+
  private:
+  scoped_refptr<base::SequencedTaskRunner> GetOrCreateTaskRunner() {
+    if (!task_runner_) {
+      task_runner_ = base::ThreadPool::CreateSequencedTaskRunner(
+          {base::MayBlock(), base::TaskPriority::USER_VISIBLE,
+           base::TaskShutdownBehavior::SKIP_ON_SHUTDOWN});
```

Comment:
Q) `SKIP_ON_SHUTDOWN` is more proper over ``CONTINUE_ON_SHUTDOWN` here?
According to the [docs](https://source.chromium.org/chromium/chromium/src/+/main:base/task/task_traits.h;l=92;bpv=1;bpt=1), it'll blocks shutdown till it's completed.